### DISCUSSION TITLE
XDrip+ Logging of Bluetooth Connection Requests?

### DISCUSSION THREAD

**Discussion by Guilden-NL** [NightscoutFoundation/xDrip#3395] [Q&A]
I connect with the G6 sensor using XDrip+ on an Android and also a Tandem pump.
Out of nowhere the pump batt started dying very fast, and my Apple that connects with it does the same. The Android running only XDrip+ (not using as a phone) is draining at 6-10 times of normal. So these battery drains out of nowhere are clearly the G6 transmitter. The transmitter as 45-50 days along when this started happening.

Does anyone know how to track connect requests in XDrip+?

I am going to see what I can find in the Android Bluetooth logs, but am hoping there is an easy way in XDrip+.  Not even going to look at the Tandem pump or Apple, as both love to hide important details from end users. ;)

**Comment by Navid200** [comment]
I used G6 with xDrip for 5 years before switching to G7 still using xDrip.
I charge my phone every other night.
With or without xDrip makes no difference.

xDrip doesn't measure how much phone battery power it uses.  But, your phone does.

If you list how much all apps use the phone battery as long as that includes all system apps as well as the screen, I don't see how xDrip could be at the top.

**Comment by Guilden-NL** [comment]
I am not saying it is XDrip+, I am just trying to remember if there is a logging function to show Bluetooth requests.
I just turned on Bluetooth HCI logging on the Android since I have it rooted and remembered that the capability is there.

So - XDrip+ on the Android.  The Android is sucking batt like crazy. Went from 80% to 0% overnight.  Normally get 2.5 days out of it. As normal,  XDrip accounts for 22.7% of batt usage as I do not use this phone for anything but a Dexcom receiver using XDrip+.  BT usage doesn't show in the normal Android apps, but I will capture it in the log.

Then I have a Tandem pump that connects to the Dexcom G6 transmitter. That is running the battery down at least twice as fast as normal.

Then to complicate things, my iPhone 13  Max Pro connects to the Tandem pump and that is running down at double or more the normal rate. Normally go a full day and then some but have charged the iPhone twice today since it went from 80% down to 38%

All have the G6 Transmitter in common and nothing else.  No other Bluetooth connections on either phone and pump.

I have been using XDrip since 2015 and used to know all of the logging, but have forgotten it since everything has been working perfectly. I believe that I will get what I need from the Android BT logging, but am still interested in XDrip+ logging if someone knows off the top of their head.  Otherwise, when I get time, I will track it down.  I used to clue in Dexcom support people as my wife used to work at Dexcom. I think I have an email I sent to them explaining all of it, just need to remember which account I was using back then. :)
